modules/transport-netty4/src/main/java/org/elasticsearch/transport/netty4/Netty4SizeHeaderFrameDecoder.java

```diff
@@ -37,17 +37,17 @@
     protected void decode(ChannelHandlerContext ctx, ByteBuf in, List<Object> out) throws Exception {
         try {
             BytesReference networkBytes = Netty4Utils.toBytesReference(in);
-            int messageLength = TcpTransport.readMessageLength(networkBytes) + HEADER_SIZE;
+            int messageLength = TcpTransport.readMessageLength(networkBytes);
+            int messageWithHeaderLength = messageLength + HEADER_SIZE;
```

Comment:
I do not even think we should be making calculations with `messageLength` if it is equal to `-1` so we should guard all of this in `if (messageLength != -1)`.

Reply:
added in: 66789927108da9aafd921a82eecc69aed0221d6e